presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/RemoveUnreferencedScalarLateralNodes.java

```diff
@@ -46,15 +46,23 @@ public PlanNode optimize(PlanNode plan, Session session, Map<Symbol, Type> types
         @Override
         public PlanNode visitLateralJoin(LateralJoinNode node, RewriteContext<PlanNode> context)
         {
-            if (node.getInput().getOutputSymbols().isEmpty() && isScalar(node.getInput())) {
-                return context.rewrite(node.getSubquery());
+            PlanNode input = node.getInput();
```

Comment:
I think we could remove this class at all since such `LateralNode` is removed in `PruneUnreferencedOutputs`